enthought_example/example_mco/tests/test_example_mco.py

```diff
@@ -43,3 +44,50 @@ def test_run(self):
             opt.run(model)
 
         self.assertEqual(mock_popen.call_count, 2)
+
+        # test whether arugment order matters on object creation
+        model.parameters = [
+            RangedMCOParameter(
+                mock.Mock(spec=RangedMCOParameterFactory),
+                initial_value=2,
+                upper_bound=3,
+                lower_bound=1,
+            )
+        ]
+
+        mock_process = mock.Mock()
+        mock_process.communicate = mock.Mock(return_value=(b"1 2 3", b""))
+
+        with mock.patch("subprocess.Popen") as mock_popen:
+            mock_popen.return_value = mock_process
+            opt.run(model)
+
+        self.assertEqual(mock_popen.call_count, 2)
+
+        # test whether arugment order matters on object creation
```

Comment:
Might be good to split this into it's own test since it was a bit of a 'hidden' bug! I know it seems kind of pointless to do this but if someone wanted to adapt these tests to their own class I think this would be an important test to have, and it might get missed if it's part of a different test

Reply:
Good point, I'll do this (and the same for any similar cases in my other pull request).